SabreTools.Hashing/SpamSum/Comparisons.cs

```diff
@@ -0,0 +1,206 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace SabreTools.Hashing.SpamSum;
+
+// 
+internal static class Comparisons
+{
+    /// <summary>
+    /// Compares how similar two SpamSums are to each other. Implements ssdeep's fuzzy_compare.
+    /// </summary>
+    /// <param name="firstHash">First hash to compare</param>
+    /// <param name="secondHash">Second hash to compare</param>
+    /// <returns>-1 on validity failure, 0 if they're not comparable, score from 0 (least similar) to 100 (most similar) otherwise.</returns>
+    /// <link>https://github.com/ssdeep-project/ssdeep/blob/df3b860f8918261b3faeec9c7d2c8a241089e6e6/fuzzy.c#L860</link>
+
+    public static int FuzzyCompare(string? firstHash, string? secondHash)
```

Comment:
General comment that the naming is a bit on the verbose side but that's something that can be cleaned up with time.

Reply:
Are you asking me to fix it during this PR, or sometime in the future? I can see how it's verbose, but I'm not quite sure what to change everything to.

Reply:
More of a general comment. It's good for this PR because it can help identify any potential gaps in the port.